src/Common/Telemetry/Telemetry.php

```diff
@@ -594,11 +594,24 @@ public function calculate_optin_status( $opted = null ) {
 	 * Calculate the optin status for the TEC plugins from various sources.
 	 *
 	 * @since 5.1.1.1
+	 * @since TBD Updated to use filter for onboarding wizard status.
 	 *
 	 * @return bool $show If the modal should show
 	 */
 	public static function calculate_modal_status(): bool {
-		// If we've already opted in, don't show the modal.
+		/**
+		 * Filter whether the telemetry modal should be shown based on onboarding wizard status.
```

Comment:
```suggestion
		 * Allow plugins to hook in and short-circuit whether the telemetry modal should be shown.
```
Doesn't in the code here have anything to do with onboarding status, technically.